frontend/app/routes/public/renew/$id/terms-and-conditions.tsx

```diff
@@ -51,6 +56,26 @@ export async function action({ context: { session }, request, params }: ActionFu
     throw new Response('Invalid CSRF token', { status: 400 });
   }
 
+  const consentSchema = z.object({
+    acknowledgeTerms: z.boolean().refine((val) => val === true, t('renew:terms-and-conditions.checkboxes.error-message.acknowledge-terms-required')),
+    acknowledgePrivacy: z.boolean().refine((val) => val === true, t('renew:terms-and-conditions.checkboxes.error-message.acknowledge-privacy-required')),
+    shareData: z.boolean().refine((val) => val === true, t('renew:terms-and-conditions.checkboxes.error-message.share-data-required')),
+  });
+
+  const data = {
+    acknowledgeTerms: formData.get('acknowledgeTerms') === 'yes',
+    acknowledgePrivacy: formData.get('acknowledgePrivacy') === 'yes',
+    shareData: formData.get('shareData') === 'yes',
+  };
```

Comment:
Consider defining an `enum` for the possible values of the checkboxes to avoid multiple string comparisons. 

If for whatever reason a future developer wants to change the checkbox value, they could just change the `enum` value rather than having to change both this `data` object and the `<InputCheckbox />`

We can then also use Zod's `nativeEnum(..)` to perform validation (there's examples of the use of this throughout the application).